flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/PartitionTestUtils.java

```diff
@@ -158,4 +159,11 @@ public ResultPartitionID getResultPartitionID() {
 			true,
 			releaseType);
 	}
+
+	public static void writeBuffers(ResultPartition partition, int numberOfBuffers, int bufferSize) throws IOException {
+		for (int i = 0; i < numberOfBuffers; i++) {
+			partition.addBufferConsumer(createFilledBufferConsumer(bufferSize, bufferSize), 0);
+		}
+		partition.finish();
+	}
```

Comment:
Actually it could be done and I indeed ever considered these parts while implementation.
It might be a separate hotfix commit for refactoring these previous tests and I did not want to touch them at that moment in limited time to block release.

Reply:
In addition, these three tests are focusing on different classes actually, which are `ResultPartition`, `ResultSubpartition` and `BoundedData`, so they are not totally duplications. If we want to unify them, we have to also refactor the relevant unit tests. I do not think it is necessary to do so now.